job.go

```diff
@@ -19,11 +23,46 @@ type JobList struct {
 }
 
 //IsJobRunning returns a int (1 - running) (0 - not)
-func IsJobRunning(job JobList) int {
+func IsJobRunning(job Job) int {
 
 	if job.State == "r" {
 		return 1
 	}
 
 	return 0
 }
+
+//GetJobs returns a slice of only jobs from both scheduled and unscheduled queues
+func GetJobs() (JobList, error) {
+	var jobs []Job
+
+	ji, err := GetJobInfo()
+
+	if err != nil {
+		return []Job{}, err
+	}
+
+	//Add running jobs to the slice first
+	for _, q := range ji.QueueInfo.Queues {
+		jobs = append(jobs, q.JobList...)
+	}
+
+	//Add pending jobs
+	jobs = append(jobs, ji.PendingJobs.JobList...)
+
+	return jobs, nil
+}
+
+//Filter allows for the passage of any function taking a JobList and Filtering its contents down.
+//Should be usable in fluent fashion as long as JobList is being returned
+func (jl JobList) Filter(filter func(j Job) bool) JobList {
```

Comment:
@dpastoor Is this something closer to what you were thinking? If you look in the filters namespace, I have some factory-esque methods that can be used this way. They're fluent chainable so it's easy to just chain together simple, generated rules _OR_ write a larger, more complicated function to handle complex logic scenarios.